src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -271,7 +272,7 @@ abstract class BCodeHelpers extends BCodeIdiomatic with BytecodeWriters {
           //
           // However, due to https://github.com/scala/scala-dev/issues/126, this currently does not
           // work, the abstract accessor for O will be marked effectivelyFinal.
```

Comment:
Yep. I think I fixed it twice actually :-) once by removing lateDeferred and once by tweaking isEffectivelyFinal